[Rust] Configurable default features from reqwest dependency

JMLX42 opened the pull request:
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work and description provides validation details.
- [x] Run build and sample generation scripts:
  ```
  ./mvnw clean package || exit
  ./bin/generate-samples.sh ./bin/configs/*.yaml || exit
  ./bin/utils/export_docs_generators.sh || exit
  ```
- [x] File PR against the correct branch: `master`
- [x] If solving a reported issue, reference it using GitHub's linking syntax (fixes #21933)
- [x] For language-specific PRs, @mention relevant technical committee members: @frol @farcaller @richardwhiuk @paladinzh @jacob-pro

## Description

This PR removes the default features (specifically `native-tls`) from the reqwest dependency in the Rust client generator. This allows users to explicitly choose their TLS backend without automatically including `openssl-sys` as a transitive dependency.

## Problem
As described in #21933, the current implementation sets `default = ["native-tls"]`, which:
- Always pulls in `openssl-sys` as a transitive dependency
- Can cause build issues for certain targets like Android
- May include an "unsafe SSL implementation" that users don't need

## Solution
- Modified `Cargo.mustache` template to set `default = []` instead of `default = ["native-tls"]`
- Users can now explicitly enable their preferred TLS backend using feature flags

## Available TLS Backends
After this change, users can explicitly select:
- `native-tls`: Uses platform-native TLS (OpenSSL on Linux, SChannel on Windows, Security Framework on macOS)
- `rustls-tls`: Uses rustls, a pure Rust TLS implementation  
- `default-tls`: Uses reqwest's default choice

## Breaking Change Notice

~~⚠️ This is a breaking change. Users will need to explicitly enable a TLS backend feature in their `Cargo.toml`:~~

This is **not** a breaking change (cf https://github.com/OpenAPITools/openapi-generator/pull/22041#issuecomment-3353095452)

## Testing
- Built the project successfully with Maven
- Regenerated all Rust samples
- Verified that generated `Cargo.toml` files have empty default features list
- Exported documentation

Fixes #21933

JMLX42 commented:
@frol @farcaller @richardwhiuk @paladinzh @jacob-pro @dsteeley for review

JMLX42 commented:
> what about adding an option to let users choose?

An option to set a Cargo feature? Sounds a bit complicated to me.

But if that's a show stopper for 7.16 then yes, I'll do it.

wing328 commented:
thanks again for putting together the following to higlight it's a breaking change:


## Breaking Change Notice
⚠️ This is a breaking change. Users will need to explicitly enable a TLS backend feature in their `Cargo.toml`:
```toml
[dependencies]
my_api_client = { version = "1.0", features = ["native-tls"] }
# or
my_api_client = { version = "1.0", features = ["rustls-tls"] }
```

is it correct to say that users need to manually update the auto-generated Cargo.toml in order to consume the API client?

JMLX42 commented:
I've updated the PR to address the feedback. Here's what changed:

## Changes Made

Added a new generator option `reqwestDefaultFeatures` that allows users to configure the default Cargo features for the reqwest dependency.

### Key Points

✅ **Backward Compatible**: Default value is `["native-tls"]` - maintains existing behavior  
✅ **Flexible**: Supports array of strings for forward compatibility  
✅ **Meets v7.17.0 criteria**: Breaking change with fallback

### Usage

**YAML configuration:**
```yaml
additionalProperties:
  reqwestDefaultFeatures: []  # No default features
  # or
  reqwestDefaultFeatures: ["rustls-tls"]  # Use rustls
  # or  
  reqwestDefaultFeatures: ["native-tls", "cookies"]  # Multiple features
```

**CLI:**
```bash
--additional-properties=reqwestDefaultFeatures=  # Empty for no defaults
--additional-properties=reqwestDefaultFeatures=rustls-tls  # Single feature
--additional-properties=reqwestDefaultFeatures=rustls-tls,cookies  # Multiple (comma-separated)
```

### Implementation Details

- Added `REQWEST_DEFAULT_FEATURES` constant and field to `RustClientCodegen`
- Default value: `Arrays.asList("native-tls")` 
- Processes both `List<String>` (from YAML) and comma-separated strings (from CLI)
- Updated `Cargo.mustache` template to use array iteration with proper comma handling
- Applied to both `reqwest` and `reqwest-trait` library sections

All builds, samples, and documentation have been regenerated successfully.

wing328 commented:
nice 👍 

thanks for the contribution
